### ISSUE TITLE
Allowing invalid TS?

### ISSUE THREAD

**Issue by mtdewulf** [cschroeter/ts-strictify#4]
Just added ts-strictify (0.4.1) to our team's typescript repo (great idea - thanks for making this open source!) and have been using it for about a day.  I noticed that I haven't seen an error yet so I ran an experiment:

I made a file "foo.ts" with the contents:

```typescript
invalid code here
```

Then ran `ts-strictify` and the file is marked as valid:

```
$ ./node_modules/.bin/ts-strictify 
🔍  Finding changed files since git revision f430c14d85957f93f95a1e874113dead431cce77
🎯  Found 1 changed files
  src/foo.ts passed
```

Note that `tsc` correctly reports errors:

```
src/foo.ts:1:9 - error TS1005: ';' expected.

1 invalid code here
          ~~~~

src/foo.ts:1:14 - error TS1005: ';' expected.

1 invalid code here
               ~~~~


Found 2 errors.
```

**Comment by cschroeter** [comment]
Mhh I try to reproduce this scenario and I get the expected behavior

```ts
// foot.ts
invalid code herevwertxe
```

```bash
demo-app on master [⇡] is 📦 v1.0.0 via ⬢ v10.16.0 
➜ yarn ts-strictify
yarn run v1.17.3
$ /Users/cschroter/Workspace/Playground/library/demo-app/node_modules/.bin/ts-strictify
🔍  Finding changed files since git revision 396701fb1d99669bf239f130ecd2d98102c8f880
🎯  Found 1 changed file
❌  src/main.ts failed
src/main.ts(1,9): error TS1005: ';' expected.
src/main.ts(1,14): error TS1005: ';' expected.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Can you provide a minimal example repo?

**Comment by mtdewulf** [comment]
If you add `foo.ts` to [this repo](https://github.com/mtdewulf/ts-strictify-bug), you should be able to reproduce.

**Comment by cschroeter** [comment]
Ah okay - so the issue here is you do not have a `tsconfig.json`. So if you run `tsc` in your project it will just print out a help list and not actually doing stuff. To fix just create a `tsconfig.json`

**Comment by mtdewulf** [comment]
I just pushed a `tsconfig.json` file and still have the same error.  Can you verify and re-open?

**Comment by mtdewulf** [comment]
```
ares:ts-strictify-bug mtdewulf$ ls
build             node_modules      package-lock.json package.json      src               tsconfig.json
ares:ts-strictify-bug mtdewulf$ cat src/foo.ts 
invalid code here
ares:ts-strictify-bug mtdewulf$ ./node_modules/.bin/ts-strictify 
🔍  Finding changed files since git revision 059872127fd03d6f0befffe2bc7cce1fa348a462
🎯  Found 1 changed file
   src/foo.ts passed
 All files passed
```

**Comment by cschroeter** [comment]
`./node_modules/.bin/ts-strictify` has issues with the current working dir. Use with yarn `yarn ts-strictify` or npm scripts ```  "scripts": {
    "check": "ts-strictify"
  },```

**Comment by mtdewulf** [comment]
Thanks, I've verified that works with the minimal repo, but I'm still having problems with my main one.  I'll keep investigating.

**Comment by mtdewulf** [comment]
Ah, I think the problem is that we have a repo with multiple sub-modules, each with their own npm config.  E.g. we have a master repo at "root", and an npm module in the directory "root/bar".  So the git paths are "root/bar/src/foo.ts", while the tsc paths are "bar/src/foo.ts".  I'm guessing this causes matching problems when identifying which files have errors?

**Comment by cschroeter** [comment]
oh this one is tricky. so an option could be to specify the path of the tsconfig, instead of assuming there is a tsconfig in the current working dir. you are using something like lerna or nx?

**Comment by mtdewulf** [comment]
The directory structure looks like:

```
root/
  .git/
  bar/
    tsconfig.json
    src/
      foo.ts
  ...
```

I wonder if instead of matching exact path, instead the suffix could be matched?  I.e. the file names reported by git will always be the same length or longer than those from tsc: `gitPath = "bar/src/foo.ts"` vs `tscPath = "src/foo.ts"`.  Instead of checking if `tscPath === gitPath`, you could do `gitPath.endsWith(tscPath)`.  Would that work?

Not using lerna or nx, just an inherited repository w/ an unusual structure.

**Comment by mtdewulf** [comment]
(I should have added that I run tsc/ts-strictify/node from the `bar/` directory.)

**Comment by RikuMantysalo** [comment]
Hey, I'm having this same issue with similar folder structure. I have code that compiles, but is not compliant with strict mode, and yet it passes.

The check also seems to take a long time, even though I have only one file with changes. I'm guessing it runs tsc for the entire project?

Would it be possible to pass in an option specifying which tsconfig to use, and then generate a temporary config which extends it, but overrides the files to include? Then it could be possible to do something like:

`npx ts-strictify --tsconfig tsconfig.json src/foo.tsx src/bar.tsx`

which would enable compatibility with `lint-staged` as well.

The tsconfig could default to the path where the script is being run from, or if not found there, traverse up the file tree until a tsconfig is found, kind of like node_modules resolve.

**Comment by cschroeter** [comment]
`ts-strictify` will resolve files in sub-modules correctly. Available in 1.5.0

**Comment by ToreJuloe** [comment]
One issue here that I encountered on the project I'm working, is that we have manually hoisted the typescript dependency in Lerna, so each package does not not have it as a direct dependency.
If typescript was installed globally with NPM, things would work more or less as expected, but without typescript, no `tsc` can be found and the files returns as " ✅ passed", hiding the fact that `tsc` didn't even run in the first place.

To check whether `tsc` failed execution on the grounds of not being there, one could check for `error.code === "ENOENT"` in the error handler.

What I found to solve my problem with typescript being hoisted, and therefore not executing, was adding the `preferLocal` parameter to `execa` like this:
#### src/lib/typescript.ts
```diff
- await execa('tsc', args, { all: true })
+ await execa('tsc', args, { all: true, preferLocal: true })
```
